### DISCUSSION TITLE
filter 도메인 이름 변경

### DISCUSSION THREAD

**Discussion by verus-j** [woowacourse-teams/2022-moamoa#72] [토론]
# 문제점
Filter라는 이름이 현재 굉장히 어색함...
```
Study - StudyFilter(?) - Filter(?) - Category
```
위의 구조인데 Filter라는 용어가 전달하는 의미가 한눈에 들어오지 않음
Filter를 사용하는 이유가 Category의 종류 중 Tag가 있어서 인데 Tag의 이름을 딴걸로 변경하여 Filter 대신 Tag를 사용하던가
아니면 Tag를 그대로 유지하고 Filter라는 이름 말고 좀 더 좋은 이름을 택해야 할 것을 보임

# 대안
Filter 보다는 Tag나 Label, Mark, Marker 등과 같은 의미를 가진 용어를 사용하는 것이 좋아 보임
[Tag 유의어](https://www.google.com/search?q=tag+synonym&sxsrf=ALiCzsY96kCrzObGX09JPCA0q8smm_BoDA%3A1658066784425&ei=YBfUYq-_GZnS2roP9ra7uA8&ved=0ahUKEwiv78uNjID5AhUZqVYBHXbbDvcQ4dUDCA4&uact=5&oq=tag+synonym&gs_lcp=Cgdnd3Mtd2l6EAMyBQgAEMsBMgUIABDLATIFCAAQywEyBQgAEMsBMgQIABAeMgQIABAeMgQIABAeMgQIABAeMgYIABAeEA8yBggAEB4QCjoFCCEQoAE6BAgjECc6EQguEIAEELEDEIMBEMcBEK8BOgUIABCABDoKCAAQgAQQhwIQFDoLCC4QgAQQxwEQrwFKBQg8EgExSgQIQRgBSgQIRhgAUJAGWMU8YJw-aAFwAHgAgAHUAYgBxgqSAQUwLjkuMZgBAKABAcABAQ&sclient=gws-wiz)

```
Study - StudyMarker - Marker - Category(Generation, Area, Tag)
Study - StudyLabel - Label - Category(Generation, Area, Tag)
Study - StudyTag - Tag - Category(Generation, Area, Label)
...
```
위와 같은 이름이 좀 더 의미를 전달하는 데 좋아보임

**Comment by nan-noo** [comment]
저도 tag대신 tech 정도가 낫지 않을까 생각 중입니다. tag는 너무 범용적이어서
태그를 나누는 카테고리 이름 -> 자바, 자스 등의 언어와 라이브러리, 프레임워크는 tech(기술) 로 가고 독서, 운동, 코드리뷰 등등은 etc(기타), 네트워크 알고리즘 등 CS관련 과목들은 cs로 하는 게 어떨까요

**Comment by jaejae-yoo** [comment]
저도 filter와 category, tag 등등 용어가 모호해서 헷갈리는 것 같아요.

`Study - StudyTag - Tag - Category(Generation, Area, Subject)`도 좋을 것 같습니다

더 직관적이면 좋을 것 같아요
